src/components/Chat/BottomSheetGif.js

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Dimensions,
+  TouchableOpacity,
+  TextInput,
+  FlatList
+} from 'react-native';
+import FastImage from 'react-native-fast-image';
+
+import {fonts, normalizeFontSize} from '../../utils/fonts';
+import {colors} from '../../utils/colors';
+import {BottomSheet} from '../BottomSheet';
+import MemoIcSearch from '../../assets/icons/Ic_search';
+import dimen from '../../utils/dimen';
+import {SIZES} from '../../utils/theme';
+import IconClear from '../../assets/icon/IconClear';
+import {getGifFeatured} from '../../service/gif';
+
+const {height: heightScreen, width: widthScreen} = Dimensions.get('window');
+
+const BottomSheetGif = React.forwardRef((props, ref) => {
+  const {onCancel, onSelect} = props;
+  const [searchText, setSearchText] = React.useState('');
+  const [list, setList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+  };
+
+  const handleOnClearText = () => {
+    setSearchText('');
+  };
+
+  const onLoad = async () => {
+    setIsLoading(true);
+    const result = await getGifFeatured();
+    if (result.code === 200) {
+      setList(result.data);
+    }
+    setIsLoading(false);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onLoad` function is making an API call but there's no error handling in case the request fails. It would be better to wrap the API call in a try-catch block and handle any potential errors.
```diff
  const onLoad = async () => {
    setIsLoading(true);
    try {
      const result = await getGifFeatured();
      if (result.code === 200) {
        setList(result.data);
      }
    } catch (error) {
      console.error(error);
    }
    setIsLoading(false);
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->